<launch>
  <arg name="GROUND_STATION_IP" default="192.168.1.143" />
  <arg name="UAV_IP" default="$(env ROS_IP)" />
  <arg name="HIGH_PORT1" default="16484" />
  <arg name="HIGH_PORT2" default="8192" />
  <arg name="LOW_PORT" default="1024" />
  <arg name="camera_nodelet_manager" default="rs_d435/realsense2_camera_manager"/>
  <arg name="robot_ns" default="hydrus"/>

  <node pkg="nodelet" type="nodelet" name="rgb_image_throttle"
        args="load jsk_topic_tools/LightweightThrottle $(arg camera_nodelet_manager)" output="screen">
    <remap from="~input"  to="rs_d435/color/image_rect_color/compressed"/>
    <remap from="~output" to="rs_d435/color/image_rect_color/throttle/compressed" />
    <param name="update_rate" value="3" />
  </node>

  <node pkg="nodelet" type="nodelet" name="depth_image_throttle"
        args="load jsk_topic_tools/LightweightThrottle $(arg camera_nodelet_manager)" output="screen">
    <remap from="~input"  to="rs_d435/aligned_depth_to_color/image_raw/compressedDepth"/>
    <remap from="~output" to="rs_d435/aligned_depth_to_color/image_raw/throttle/compressedDepth" />
    <param name="update_rate" value="3" />
  </node>

  ### UAV to GroundStation
  <node pkg="jsk_network_tools" type="silverhammer_highspeed_streamer.py" name="highspeed_streamer">
    <param name="message" value="mbzirc2020_task2_common/FC2OCSData" />
    <param name="to_ip" value="$(arg GROUND_STATION_IP)" />
    <param name="to_port" value="$(arg HIGH_PORT1)" />
    <param name="send_rate" value="5" />
    <param name="bandwidth" value="2000000" />
    <param name="packet_size" value="1000" />
  </node>

  ### GroundStation to UAV
  <node pkg="jsk_network_tools" type="silverhammer_lowspeed_receiver.py" name="lowspeed_receiver">
    <remap from="~output" to="/task_start" />
    <param name="message" value="std_msgs/Empty" />
    <param name="receive_ip" value="$(arg UAV_IP)" />
    <param name="receive_port" value="$(arg LOW_PORT)" />
    <param name="receive_buffer_size" value="1000" />
  </node>

  ### GroundStation to UAV
  <node pkg="jsk_network_tools" type="silverhammer_highspeed_receiver.py" name="receiver">
    <param name="message" value="mbzirc2020_task2_common/OCS2FCData" />
    <param name="receive_ip" value="$(arg UAV_IP)" />
    <param name="receive_port" value="$(arg HIGH_PORT2)" />
    <param name="topic_prefix" value="" />
  </node>

  <node pkg="mbzirc2020_task2_common" type="world_to_root_tf_msg_publisher.py" name="world_to_root_tf_msg_publisher">
    <param name="robot_ns" value="$(arg robot_ns)"/>
  </node>

</launch>
